### PR TITLE
Calibrate the coordinates based on frame position.

### LINKED ISSUE #2207: [BUG]: Can't drag component if there is some space on top/left with Mode Absolute
Given the following code example...

``` html
<div style="height: 100px;">
  Some Space on top or left
</div>
<div id="gjs">
  <p style="padding: 25px">
    Hesdllo World!!!</p>
</div>

```

``` js
editor.getModel().set('dmode', 'absolute');
```

Results in no ability to drag and drop elements to correct position using drag icon on the panel.

image

See Fiddle for demonstration...

https://jsfiddle.net/niisar/zc51gtnv/

### PR DESCRIPTION
**Pull request by emilsedgh**
Calibrate the coordinates based on frame position.

Since the toolbar lives outside the canvas frame, the event's generated on it
have clientX and clientY relative to the page

This patch makes sure we consider frame offset from the page. 

Fixes #2207,#2372

### CODE REVIEW

**Comment by emilsedgh** [comment]
@artf Any chance we could review and merge this?

**Comment by emilsedgh** [comment]
I updated the branch to resolve conflicts.

**Comment by emilsedgh** [comment]
Please note that the issue has been reported at least 3 times:
#2207, #2372 and #2473

File: src/dom_components/view/ToolbarButtonView.js [hunk 1]
```diff
@@ -22,9 +22,32 @@ export default Backbone.View.extend({
   handleClick(event) {
     event.preventDefault();
     event.stopPropagation();
-    const { em } = this;
```
Comment (artf) [inline src/dom_components/view/ToolbarButtonView.js]: I think you forgot this 😬

Comment (emilsedgh) [reply src/dom_components/view/ToolbarButtonView.js]: Oh right of course.
Can you explain what's the different between `this.em` and `this.editor`?

Comment (artf) [reply src/dom_components/view/ToolbarButtonView.js]: `editor` is more or less a public API layer of `em` but they both act as a Service Manager

**Comment by artf** [comment]
Thanks @emilsedgh ❤️

File: src/dom_components/view/ToolbarButtonView.js (hunk 1 above)
Comment (emilsedgh) [reply src/dom_components/view/ToolbarButtonView.js]: In that case I don't like the fact that in this patch I'm using both `em` and `editor`. I think we should use either one not both.
You agree with that? Should I update my branch or something?

**Comment by ramusesan** [comment]
when can expect to be pushed this changes to **npm repo**
@artf

**Comment by artf** [comment]
@ramusesan I'll try to find time to release it tomorrow

**Comment by ramusesan** [comment]
@artf  ok thanks, I will be waiting

**Comment by alemenciones** [comment]
What happens when i change device and the component is outside the canvas? this's not working

image

**Comment by artf** [comment]
> What happens when i change device and the component stay outside of canvas? this's not working maybe
> 
> image

The outline boxes keep track of the components even outside of their frames

**Comment by alemenciones** [comment]
I'm talking about the calibration of coordinates x/y in the `tlb-move` event, when the componente is outside of frame, this is not working good

**Pull request merged**